packages/odyssey-react-mui/src/labs/DataComponents/StackContent.tsx

```diff
@@ -0,0 +1,247 @@
+      currentRowId,
+      draggingRowId,
+      hoveredRowId,
+    }: {
+      currentRowId: string;
+      draggingRowId?: string;
+      hoveredRowId?: string;
+    }) => string | undefined;
+    handleDragHandleKeyDown: ({
+      table,
+      row,
+      event,
+    }: {
+      table: MRT_TableInstance<MRT_RowData>;
+      row: MRT_Row<MRT_RowData>;
+      event: React.KeyboardEvent<HTMLButtonElement>;
+    }) => void;
+    handleDragHandleOnDragCapture: (
+      table: MRT_TableInstance<MRT_RowData>,
+    ) => void;
+    handleDragHandleOnDragEnd: (table: MRT_TableInstance<MRT_RowData>) => void;
+    resetDraggingAndHoveredRow: (table: MRT_TableInstance<MRT_RowData>) => void;
+    updateRowOrder: ({
+      rowId,
+      newRowIndex,
+    }: {
+      rowId: string;
+      newRowIndex: number;
+    }) => void;
+  };
+  rowSelection: MRT_RowSelectionState;
+  setRowSelection: Dispatch<SetStateAction<MRT_RowSelectionState>>;
+  stackOptions: StackProps;
+  totalRows: UniversalProps["totalRows"];
+};
+
+const StackContent = ({
+  currentLayout,
+  data,
+  emptyState,
+  hasRowReordering,
+  hasRowSelection,
+  isEmpty,
+  isLoading,
+  isNoResults,
+  isRowReorderingDisabled,
+  onReorderRows,
+  pagination,
+  rowReorderingUtilities,
+  rowSelection,
+  setRowSelection,
+  stackOptions,
+  totalRows,
+}: StackContentProps) => {
+  const handleRowSelectionChange = useCallback(
+    (row: MRT_RowData) => {
+      setRowSelection((prev) => {
+        const newSelection = { ...prev };
+        if (newSelection[row.id]) {
+          delete newSelection[row.id];
+        } else {
+          newSelection[row.id] = true;
+        }
+        return newSelection;
+      });
+    },
+    [setRowSelection],
+  );
+
+  const renderDetailPanelProp = stackOptions.renderDetailPanel;
+
+  const renderDetailPanel = useCallback(
+    (row: MRT_RowData) => {
+      return renderDetailPanelProp?.({ row });
+    },
+    [renderDetailPanelProp],
+  );
+
+  const { updateRowOrder } = rowReorderingUtilities;
+
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const StackContainer = styled("div", {
+    shouldForwardProp: (prop) =>
+      prop !== "odysseyDesignTokens" &&
+      prop !== "currentLayout" &&
+      prop !== "maxGridColumns",
+  })<{
+    odysseyDesignTokens: DesignTokens;
+    currentLayout: StackLayout;
+    maxGridColumns: number;
+  }>(({ odysseyDesignTokens, currentLayout, maxGridColumns }) => ({
+    display: currentLayout === "list" ? "flex" : "grid",
+    flexDirection: "column",
+    gap: odysseyDesignTokens.Spacing5,
+
+    ...(currentLayout === "grid" && {
+      [`@media (max-width: 720px)`]: {
+        gridTemplateColumns: "repeat(1, 1fr)",
+      },
+      [`@media (min-width: 720px) and (max-width: 960px)`]: {
+        gridTemplateColumns: "repeat(2, 1fr)",
+      },
+      [`@media (min-width: 960px)`]: {
+        gridTemplateColumns: `repeat(${maxGridColumns}, 1fr)`,
+      },
+    }),
+  }));
```

Comment:
Can we add the design tokens context consumer to the top of the component? Just to keep that as a consistent pattern. It doesn't need to be this far down necessarily, but I do like that it's around the components we're creating.

Still, I don't think we should be creating styled components inline like this. I believe theres a cost to doing so, and React can't properly memoize them.

I'm not 100% sure, but at first glance, my opinion is that you'd have to memoize these styled components with `useMemo`.

Reply:
I do agree any styled components should be created outside of the component function. Just need to move them outside of `StackContent` and everything should be fine